src/VisualStudio/LiveShare/Impl/RunCodeActionsHandler.cs

```diff
@@ -0,0 +1,118 @@
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using Newtonsoft.Json.Linq;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+{
+    /// <summary>
+    /// Run code actions handler.  Called when lightbulb invoked.
+    /// Code actions must be applied from the UI thread in VS.
+    /// </summary>
+    internal class RunCodeActionsHandler : CodeActionsHandlerBase, ILspRequestHandler<LSP.ExecuteCommandParams, object, Solution>
+    {
+        private IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        public RunCodeActionsHandler(ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, IThreadingContext threadingContext)
+            : base(codeFixService, codeRefactoringService)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<object> HandleAsync(LSP.ExecuteCommandParams request, RequestContext<Solution> requestContext, CancellationToken cancellationToken)
+        {
+            // Unwrap the command to get the RunCodeActions command.
+            if (request.Command.StartsWith(CodeActionsHandlerShim.RemoteCommandNamePrefix))
+            {
+                var command = ((JObject)request.Arguments[0]).ToObject<LSP.Command>();
+                request.Command = command.CommandIdentifier;
+                request.Arguments = command.Arguments;
+            }
+
+            if (request.Command == RunCodeActionCommandName)
+            {
+                var runRequest = ((JToken)request.Arguments.Single()).ToObject<RunCodeActionParams>();
+                var codeActions = await GetCodeActionsAsync(requestContext.Context,
+                    runRequest.CodeActionParams.TextDocument.Uri,
+                    runRequest.CodeActionParams.Range,
+                    cancellationToken).ConfigureAwait(false);
+
+                var actionToRun = codeActions?.FirstOrDefault(a => a.Title == runRequest.Title);
+
+                if (actionToRun != null)
+                {
+                    // TODO - This UI thread dependency must be removed.
+                    // https://github.com/dotnet/roslyn/projects/45#card-20619668
+                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+                    foreach (var operation in await actionToRun.GetOperationsAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
❗️ This should be `ConfigureAwait(true)`

Reply:
➡️ Resolved by moving `SwitchToMainThreadAsync` inside the loop.

Reply:
discussed offline - I made the change to move the switch inside the loop, so now it shouldn't need ConfigureAwait(true) here.